service/backend/pchain/indexer/parser.go

```diff
@@ -215,10 +205,6 @@ func (p *parser) parseBlockBytes(proposerBytes []byte) (*ParsedBlock, error) {
 		return nil, fmt.Errorf("fetching proposer from block bytes errored with %w", err)
 	}
 
-	if p.genesisTimestamp.IsZero() {
```

Comment:
`p.genesisTimestamp` is initialized upon construction and update upon genesis parsing. It cannot be zero